sdk/keyvault/keyvault-certificates/src/index.ts

```diff
@@ -870,44 +876,56 @@ export class CertificatesClient {
 
   /**
    * Creates a new certificate. If this is the first version, the certificate resource is created. This operation requires the certificates/create permission.
+   * This function returns a Long Running Operation poller that allows you to wait indifinetly until the certificate is signed.
    *
    * Example usage:
    * ```ts
    * const client = new CertificatesClient(url, credentials);
-   * await client.createCertificate("MyCertificate", {
+   * const poller = await client.beginCreateCertificate("MyCertificate", {
    *   issuerName: "Self",
    *   subjectName: "cn=MyCert"
    * });
+   *
+   * // Serializing the poller
+   * const serialized = poller.toJSON();
+   * // A new poller can be created with:
+   * // await client.beginCreateCertificate("MyCertificate", policy, { resumeFrom: serialized });
+   *
+   * // Waiting until it's done
+   * const certificate = await poller.pollUntilDone();
+   * console.log(certificate);
    * ```
-   * @summary Creates a certificate
+   * @summary Creates a certificate through a Long Running Operation Poller
    * @param name The name of the certificate
    * @param certificatePolicy The certificate's policy
-   * @param [options] Optional parameters
+   * @param [createCertificateOptions] Optional parameters to the createCertificate operation
+   * @param [pollerOptions] Optional parameters to the creation of the poller
    */
-  public async createCertificate(
+  public async beginCreateCertificate(
     name: string,
     certificatePolicy: CertificatePolicy,
-    options: CreateCertificateOptions = {}
-  ): Promise<Certificate> {
-    const span = this.createSpan("createCertificate", options);
-
-    let result: CreateCertificateResponse;
-
-    try {
-      result = await this.client.createCertificate(this.vaultBaseUrl, name, {
-        ...this.setParentSpan(span, options.requestOptions || {}),
-        certificateAttributes: {
-          ...options.certificateAttributes,
-          enabled: options.enabled
-        },
-        tags: options.tags,
-        certificatePolicy: toCorePolicy(certificatePolicy)
-      });
-    } finally {
-      span.end();
-    }
+    createCertificateOptions: BeginCreateCertificateOptions = {}
+  ): Promise<CreateCertificatePoller> {
+    // Making a fake client to reduce the duplicated code
+    const pollerKeyVaultClient: CertificatesClientInterface = {
+      createCertificate: this.createCertificate.bind(this),
+      getPlainCertificateOperation: this.getPlainCertificateOperation.bind(this),
+      cancelCertificateOperation: this.cancelCertificateOperation.bind(this),
+      getCertificateWithPolicy: this.getCertificateWithPolicy.bind(this),
+      getCertificatePolicy: this.getCertificatePolicy.bind(this)
+    };
 
-    return this.getCertificateFromCertificateBundle(result);
+    const poller = new CreateCertificatePoller({
+      client: pollerKeyVaultClient,
```

Comment:
per my other review, I think you can pass `this` as the value since it still satisfies the interface, it's just polymorphic downcasting.